src/ClientData/CaptureData.cpp

```diff
@@ -251,4 +252,25 @@ void CaptureData::UpdateTimerDurations() {
   return result;
 }
 
+[[nodiscard]] std::optional<ThreadStateSliceInfo>
+CaptureData::FindThreadStateSliceInfoFromTimestamp(int64_t thread_id, uint64_t timestamp) {
+  absl::MutexLock lock{&thread_state_slices_mutex_};
+  if (thread_state_slices_.count(thread_id) == 0u) {
+    return std::nullopt;
+  }
+
+  std::vector<ThreadStateSliceInfo>& thread_state_bar = thread_state_slices_[thread_id];
+  auto slice = std::lower_bound(thread_state_bar.begin(), thread_state_bar.end(),
+                                timestamp,  // compare based on ending timestamps
+                                [](const ThreadStateSliceInfo& a, const uint64_t& b) -> bool {
```

Comment:
Is it faster?

Reply:
I don't think it would be fast in this situation. But in some situations it is actually faster because it avoids potential pointer aliasing which means the compiler can do better optimizations. Passing by value has basically the same effect as the `restrict` keyword on pointer arguments in C (See https://en.wikipedia.org/wiki/Restrict for reference)